src/neo-vm/ExecutionEngine.cs

```diff
@@ -1189,19 +995,50 @@ protected virtual void LoadContext(ExecutionContext context)
             if (InvocationStack.Count >= MaxInvocationStackSize)
                 throw new InvalidOperationException();
             InvocationStack.Push(context);
+            if (EntryContext is null) EntryContext = context;
+            CurrentContext = context;
         }
 
         public ExecutionContext LoadScript(Script script, int rvcount = -1)
         {
-            ExecutionContext context = new ExecutionContext(script, CurrentContext?.Script, rvcount);
+            ExecutionContext context = new ExecutionContext(script, CurrentContext?.Script, rvcount, ReferenceCounter);
             LoadContext(context);
             return context;
         }
 
         protected virtual bool OnSysCall(uint method) => false;
 
-        protected virtual bool PostExecuteInstruction(Instruction instruction) => true;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public StackItem Peek(int index = 0)
+        {
+            var stack = CurrentContext?.EvaluationStack ?? ResultStack;
+            return stack.Peek(index);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public StackItem Pop()
+        {
+            return CurrentContext.EvaluationStack.Pop();
+        }
+
+        protected virtual bool PostExecuteInstruction(Instruction instruction)
+        {
+            ReferenceCounter.CheckZeroReferred();
```

Comment:
if this method return the count, we can save one call (Count) but it's good to me like this too

Reply:
Changed.